common/src/View/EntityAttributeGrid.cpp

```diff
@@ -227,18 +231,40 @@ namespace TrenchBroom {
         class EntityAttributeCellEditor : public wxGridCellTextEditor
         {
         private:
+            EntityAttributeGrid* m_grid;
             EntityAttributeGridTable* m_table;
+            int m_row, m_col;
             
         public:
-            EntityAttributeCellEditor(EntityAttributeGridTable* table)
-            : m_table(table) {}
+            EntityAttributeCellEditor(EntityAttributeGrid* grid, EntityAttributeGridTable* table)
+            : m_grid(grid),
+            m_table(table),
+            m_row(-1),
+            m_col(-1) {}
 
+        private:
+            void OnCharHook(wxKeyEvent& event) {
+                if (event.GetKeyCode() == WXK_TAB) {
+                    // HACK: Consume tab key and use it for cell navigation.
+                    // Otherwise, wxTextCtrl::AutoComplete uses it for cycling between completions (on Windows)
+                    m_grid->tabNavigate(m_row, m_col, !event.ShiftDown());
+                    return;
+                }
+                event.Skip();
+            }
+
+        public:
             void BeginEdit(int row, int col, wxGrid* grid) override {
                 wxGridCellTextEditor::BeginEdit(row, col, grid);
+
+                m_row = row;
+                m_col = col;
                 
                 wxTextCtrl *textCtrl = Text();
                 ensure(textCtrl != nullptr, "wxGridCellTextEditor::Create should have created control");
 
+                textCtrl->Bind(wxEVT_CHAR_HOOK, &EntityAttributeCellEditor::OnCharHook, this);
```

Comment:
I'm unsure about multiple calls to Bind with the same arguments. The wxWidgets docs don't seem to specify what happens in that case, so it may or may not work. I'd feel better if you protect against multiple calls somehow, or if you could unbind the hook in EndEdit.